Sources/App/Extensions/Date+Utils.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Copyright © 2017 Patrick Balestra. All rights reserved.
+//
+
+import Vapor
+
+extension Date {
```

Comment:
This seems to work fine, but maybe we can leverage the already provided APIs in `NSCalendar` and `NSDate`. Take a look at this function, it seems to do exactly what you need: https://gist.github.com/minorbug/468790060810e0d29545